externs/browser/chrome.js

```diff
@@ -76,29 +76,170 @@ function ChromeEvent() {}
 /**
  * @param {!Function} callback
  * @return {undefined}
+ * @see https://developer.chrome.com/extensions/events#method-Event-addListener
  */
 ChromeEvent.prototype.addListener = function(callback) {};
 
 
 /**
  * @param {!Function} callback
  * @return {undefined}
+ * @see https://developer.chrome.com/extensions/events#method-Event-removeListener
  */
 ChromeEvent.prototype.removeListener = function(callback) {};
 
 
 /**
  * @param {!Function} callback
  * @return {boolean}
+ * @see https://developer.chrome.com/extensions/events#method-Event-hasListener
  */
 ChromeEvent.prototype.hasListener = function(callback) {};
 
 
-/** @return {boolean} */
+/**
+ * @return {boolean}
+ * @see https://developer.chrome.com/extensions/events#method-Event-hasListeners
+ */
 ChromeEvent.prototype.hasListeners = function() {};
 
-// TODO(tbreisacher): Add the addRules, getRules, and removeRules methods?
 
+/**
+ * @param {!Array<Rule>} rules
+ * @param {function(!Array<Rule>): void=} callback
```

Comment:
I think you should enforce it (`!Array<!Rule>`), since a null rule makes no sense.
If you don't enforce it, it's easier for bad code to slip in and it becomes harder to enforce it later.